x-pack/plugins/security_solution/public/flyout/shared/components/expandable_panel.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback } from 'react';
+import {
+  EuiButtonIcon,
+  EuiSplitPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiIcon,
+  EuiLink,
+  EuiTitle,
+  EuiText,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import styled from 'styled-components';
+
+const StyledEuiFlexItem = styled(EuiFlexItem)`
+  margin-right: ${({ theme }) => theme.eui.euiSizeM};
+`;
+
+const StyledEuiIcon = styled(EuiIcon)`
+  margin: ${({ theme }) => theme.eui.euiSizeS} 0;
+`;
+
+const StyledEuiLink = styled(EuiLink)`
+  font-size: 12px;
```

Comment:
I learned about this from rules management team: EUI team would encourage moving to Emotion or use Emotion in the new code as minimum. There are issues though related to inability to use styled-components and @emotion/react in one plugin but @emotion/css comes to rescue. You can check it out here https://github.com/elastic/eui/discussions/6828.

eui theme has prebuilt values for font size and font weight:
- font size of 12px is `euiFontSizeXS`
- font weight of 700 is `euiFontWeightBold` see [here](https://elastic.github.io/eui/#/theming/typography/values#font-weight)

Reply:
great point on using emotion over styled component, and even better point using eui variables, thank you!